puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -159,8 +159,13 @@ def run_duration
   # @api private
   def event_to_hash(event)
     corrective_change = defined?(event.corrective_change) ? event.corrective_change : nil
+
+    # PUP-1542 sometimes causes status to be nil; fix it up here so we
+    # can store the rest of the report
+    status = event.status || 'failure'
```

Comment:
We've walked through the code where this can happen; it's a pretty narrow case involving an exception being raised inside a rescue block. So I'm confident that this is indeed a failure. But a note to take out the workaround when (if) we fix it is a good idea, I'll do that.